src/java.base/windows/classes/sun/nio/fs/WindowsPath.java

```diff
@@ -831,12 +831,57 @@ long openForReadAttributeAccess(boolean followLinks)
         int flags = FILE_FLAG_BACKUP_SEMANTICS;
         if (!followLinks)
             flags |= FILE_FLAG_OPEN_REPARSE_POINT;
+        try {
+            return openFileForReadAttributeAccess(flags);
+        } catch (WindowsException e) {
+            if (!followLinks || e.lastError() != ERROR_CANT_ACCESS_FILE)
+                throw e;
+            // Object could be a Unix domain socket
+            long handle = openSocketForReadAttributeAccess();
+            if (handle == INVALID_HANDLE_VALUE)
+                throw e;
+            return handle;
+        }
+    }
+
+    private long openFileForReadAttributeAccess(int flags)
+        throws WindowsException
+    {
         return CreateFile(getPathForWin32Calls(),
-                          FILE_READ_ATTRIBUTES,
-                          (FILE_SHARE_READ | FILE_SHARE_WRITE | FILE_SHARE_DELETE),
-                          0L,
-                          OPEN_EXISTING,
-                          flags);
+                            FILE_READ_ATTRIBUTES,
+                            (FILE_SHARE_READ | FILE_SHARE_WRITE | FILE_SHARE_DELETE),
+                            0L,
+                            OPEN_EXISTING,
+                            flags);
+    }
+
+    /**
+     * Returns INVALID_HANDLE_VALUE if path is not a socket
+     * and a handle to the socket file if it is.
+     */
+    private long openSocketForReadAttributeAccess() {
```

Comment:
The methods here throw WindowsException when they fail and might be better to keep it consistent and throw WindowsException rather than returning INVALID_HANDLE_VALUE.

Reply:
Okay

Reply:
Actually, there's a mistake in the current latest version (04). Should have tested it before pushing. Will push update shortly.